`pl.read_excel` fails with `invalid parameters: schema_sample_rows cannot be 0`

### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import polars as pl
df_read = pl.read_excel(excel_bytes, engine="calamine", infer_schema_length=0)
```

OR

pytest -v ../polars/py-polars/tests/unit/io/test_spreadsheet.py::test_write_excel_bytes[calamine]



### Log output

```shell
../polars/py-polars/tests/unit/io/test_spreadsheet.py::test_write_excel_bytes[calamine] FAILED                                                                                                                                                        [100%]

========================================================================================================================= FAILURES ==========================================================================================================================
_____________________________________________________________________________________________________________ test_write_excel_bytes[calamine] ______________________________________________________________________________________________________________
../polars/py-polars/tests/unit/io/test_spreadsheet.py:485: in test_write_excel_bytes
    df_read = pl.read_excel(excel_bytes, engine=engine, infer_schema_length=0)
../polars/py-polars/polars/_utils/deprecation.py:92: in wrapper
    return function(*args, **kwargs)
../polars/py-polars/polars/_utils/deprecation.py:92: in wrapper
    return function(*args, **kwargs)
../polars/py-polars/polars/io/spreadsheet/functions.py:381: in read_excel
    _read_spreadsheet(
../polars/py-polars/polars/io/spreadsheet/functions.py:659: in _read_spreadsheet
    name: reader_fn(
../polars/py-polars/polars/io/spreadsheet/functions.py:1097: in _read_spreadsheet_calamine
    ws_arrow = parser.load_sheet_eager(sheet_name, **read_options)
../.conda/envs/rapids/lib/python3.12/site-packages/fastexcel/__init__.py:396: in load_sheet_eager
    return self._reader.load_sheet(
E   _fastexcel.InvalidParametersError: invalid parameters: schema_sample_rows cannot be 0, as it would prevent dtype inferring
------------------------------------------------------------------------------------------------------------------- Captured stderr call --------------------------------------------------------------------------------------------------------------------
run FilterExec
dataframe filtered
run ProjectionExec
run ProjectionExec
run ProjectionExec
================================================================================================================== short test summary info ==================================================================================================================
FAILED ../polars/py-polars/tests/unit/io/test_spreadsheet.py::test_write_excel_bytes[calamine] - _fastexcel.InvalidParametersError: invalid parameters: schema_sample_rows cannot be 0, as it would prevent dtype inferring
```


### Issue description

We encountered this bug running the polars test suite in cudf_polars. It looks like fastexcel no longer allows `schema_sample_rows=0` ([ref PR](https://github.com/ToucanToco/fastexcel/pull/304)).

And the line where we set `schema_sample_rows` is here: https://github.com/pola-rs/polars/blob/main/py-polars/polars/io/spreadsheet/functions.py#L733

### Expected behavior

I think we should make the test xfail when the fastexcel version is 0.12.1 or remove the case where `schema_sample_rows=0` from the failing polars test. I'd be happy to open a PR to fix this.

### Installed versions

<details>

```
--------Version info---------
Polars:              1.19.0
Index type:          UInt32
Platform:            Linux-5.4.0-182-generic-x86_64-with-glibc2.35
Python:              3.12.8 | packaged by conda-forge | (main, Dec  5 2024, 14:24:40) [GCC 13.3.0]
LTS CPU:             False

----Optional dependencies----
Azure CLI            <not installed>
adbc_driver_manager  1.3.0
altair               5.5.0
azure.identity       <not installed>
boto3                1.35.88
cloudpickle          3.1.0
connectorx           0.4.0
deltalake            0.23.2
fastexcel            0.12.1
fsspec               2024.12.0
gevent               24.11.1
google.auth          2.37.0
great_tables         0.15.0
matplotlib           3.10.0
nest_asyncio         1.6.0
numpy                2.0.2
openpyxl             3.1.5
pandas               2.2.3
pyarrow              18.1.0
pydantic             2.10.5
pyiceberg            <not installed>
sqlalchemy           2.0.36
torch                2.5.1.post303
xlsx2csv             0.8.4
xlsxwriter           3.2.0
```

</details>

--- a/py-polars/polars/io/spreadsheet/functions.py
+++ b/py-polars/polars/io/spreadsheet/functions.py
@@ -1114,8 +1114,12 @@ def _read_spreadsheet_calamine(
     schema_overrides = schema_overrides or {}
     if read_options.get("schema_sample_rows") == 0:
         # ref: https://github.com/ToucanToco/fastexcel/issues/236
-        read_options["dtypes"] = dict.fromkeys(range(16384), "string")
-
+        del read_options["schema_sample_rows"]
+        read_options["dtypes"] = (
+            "string"
+            if fastexcel_version >= (0, 12, 1)
+            else dict.fromkeys(range(16384), "string")
+        )
     elif schema_overrides and fastexcel_version >= (0, 10):
         parser_dtypes = read_options.get("dtypes", {})
         for name, dtype in schema_overrides.items():

